### PR TITLE
Update BaseLink.d.ts

### PR DESCRIPTION
**Pull request by paibamboo**
inputMode is not compatible with @types/react@16.9.11

### CODE REVIEW

File: packages/react-router5/types/BaseLink.d.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ declare class BaseLink extends Component<BaseLinkProps, BaseLinkState> {
             style?: React.CSSProperties
             tabIndex?: number
             title?: string
-            inputMode?: string
+            inputMode?: 'none' | 'text' | 'tel' | 'url' | 'email' | 'numeric' | 'decimal' | 'search'
```
Comment (troch) [inline packages/react-router5/types/BaseLink.d.ts]: Is there a way to import this type from React rather than maintaining it manually?

Comment (JSteunou) [reply packages/react-router5/types/BaseLink.d.ts]: could it be ? `inputMode?: React.HTMLAttributes<React.HTMLElement>`

Comment (JSteunou) [reply packages/react-router5/types/BaseLink.d.ts]: Ok find a way: remove `inputMode` declaration and instead pick it from React

```ts
    render(): React.DetailedReactHTMLElement<
        Pick<React.HTMLAttributes<HTMLElement>, 'inputMode'> & {
            href: any
            className: string
     ...
```

Comment (JSteunou) [reply packages/react-router5/types/BaseLink.d.ts]: And I think a lot of attributes could be picked from `React`

Comment (titouancreach) [reply packages/react-router5/types/BaseLink.d.ts]: Another perspective: Do we need to be as much accurate as we are? In which case this is useful for the consumer of this component to have those informations?

I suggest to replace `React.DetailedReactHtmlElement<.......>`  with `React.ReactNode`.

Comment (paibamboo) [reply packages/react-router5/types/BaseLink.d.ts]: Feedback updated. I agreed with @titouancreach suggestion but opted for a smaller change for now.

**Comment by JSteunou** [comment]
anyone up for a release ? @troch ? @titouancreach ?

**Comment by troch** [comment]
@paibamboo I didn't realise you modified directly the types. Can you make the change to `BaseLink.ts` directly instead?

**Comment by paibamboo** [comment]
@troch In `BaseLink.ts` line 101 It's just `render() {` without specifying type. I think `BaseLink.d.ts` isn't auto-generated. If it was, BaseLink's `render()` return type would be defaulted to `ReactNode` because the class extends `React.Component`. So that means modifying `BaseLink.ts` wouldn't fix the issue. Correct me if I'm wrong.

**Comment by JSteunou** [comment]
@troch can you please release this fix? You have a good MR, reviews / approvals...

**Comment by tonmanayo** [comment]
@troch we'd really appreciate this too

**Comment by katywings** [comment]
Meanwhile @types/react is already at 16.9.17, please finish this or let us help you to get it done :)

**Comment by PNNutkung** [comment]
Let Get This Merge

**Comment by taddj** [comment]
when will this be released?

**Comment by nigelthinksprint** [comment]
Could this please get merged in ?

**Comment by troch** [comment]
Hello, sorry for the silence. This bug will be fixed as part of a general maintenance (#446). ETA is within two weeks.

**Comment by pbadenski** [comment]
Any way we can help to get this over the line?

**Comment by troch** [comment]
Closing because fixed as part of #446

**Pull request closed without merging**